drivers/hopwa_caper/app/models/hopwa_caper/generators/fy2024/generator.rb

```diff
@@ -0,0 +1,186 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# run manually with
+# Reporting::Hud::RunReportJob.new.perform(HopwaCaper::Generators::Fy2024::Generator.name, HudReports::ReportInstance.last, email:false)
+module HopwaCaper::Generators::Fy2024
+  class Generator < ::HudReports::GeneratorBase
+    def self.fiscal_year
+      'FY 2024'
+    end
+
+    def self.generic_title
+      'HOPWA CAPER'
+    end
+
+    def self.short_name
+      'HOPWA CAPER'
+    end
+
+    def self.file_prefix
+      "v1.0 #{short_name} #{fiscal_year}"
+    end
+
+    def self.default_project_type_codes
+      HudUtility2024.path_project_type_codes
+    end
+
+    def initialize(report)
+      super(report)
+      return unless report&.persisted?
+
+      reset_report
+      build_hopwa_caper_models
+    end
+
+    def reset_report
+      report.hopwa_caper_enrollments.delete_all
+      report.hopwa_caper_services.delete_all
+      report.report_cells.with_deleted.destroy_all
+      report.update! state: 'Waiting'
+    end
+
+    def url
+      hud_reports_hopwa_caper_url(report, { host: ENV['FQDN'], protocol: 'https' })
+    end
+
+    def self.questions
+      [
+        HopwaCaper::Generators::Fy2024::Sheets::DemographicsAndPriorLivingSituationSheet,
+        HopwaCaper::Generators::Fy2024::Sheets::TbraSheet,
+        HopwaCaper::Generators::Fy2024::Sheets::StrmuSheet,
+        HopwaCaper::Generators::Fy2024::Sheets::PhpSheet,
+      ].map do |q|
+        [q.question_number, q]
+      end.to_h.freeze
+    end
+
+    def self.valid_question_number(question_number)
+      return question_number
+    end
+
+    def self.filter_class
+      ::Filters::HudFilterBase
+    end
+
+    def self.allowed_options
+      [
+        :start,
+        :end,
+        :coc_codes,
+        :project_ids,
+        :data_source_ids,
+        :project_type_codes,
+        :project_group_ids,
+      ]
+    end
+
+    protected
+
+    def service_history_enrollments
+      overlapping_enrollments = GrdaWarehouse::ServiceHistoryEnrollment.entry.
+        open_between(start_date: @report.start_date, end_date: @report.end_date)
+
+      # tbra has a 15 year look-back
+      look_back = 15.years
+      scope = GrdaWarehouse::ServiceHistoryEnrollment.entry.
+        open_between(start_date: @report.start_date - look_back, end_date: @report.end_date).
+        where(client_id: overlapping_enrollments.select(:client_id))
+
+      @filter = self.class.filter_class.new(
+        user_id: @report.user_id,
+        enforce_one_year_range: false,
+      ).update(@report.options)
+      @filter.funder_ids = HudUtility2024.funder_components.fetch('HUD: HOPWA')
+
+      @filter.project_ids = @report.project_ids
+      @filter.apply(scope)
+    end
+
+    def build_hopwa_caper_models
```

Comment:
I haven't been able to find the `def run!` method that is being called.  Whatever is called from the `RunReportJob` DJ should set the `state` column of the report instance to `Started` and the `started_at` timestamp when the report starts.  (I'm not seeing the running report reflected correctly in the UI, it just says `Queued` but it _is_ running.

Reply:
This is working slightly better now, it shows "Started" when it starts, and has the start time, but when it completes it's filling the `started_at` timestamp with a new value making it look like it only took a few seconds to run.  `completed_at` is getting correctly filled, but I think it's updating both columns and should only update `completed_at` when it completes.

Reply:
oh yeah, looks like it resets the started_at on the first question. I adjusted it so that prepare_report marks it as started instead of in report.start(question...)